crates/prune/src/pruner.rs

```diff
@@ -145,9 +151,10 @@ impl<DB: Database> Pruner<DB> {
 
         let mut done = true;
         let mut stats = PrunerStats::new();
+        let mut deleted = 0;
 
         for (segment, purpose) in segments {
-            if delete_limit == 0 {
+            if limit.at_limit(start, deleted) {
```

Comment:
we always pass the same `start`, can we make it a field of the `PruneLimit` struct? So only the `deleted` input argument will change